Introduce prototype-style function definitin for generic memchr.
Upstream did this as part of:

commit 1570a72bb8359c9a3a64c4e22d9ea47ce6fcff31
Author: Will Newton <will.newton@linaro.org>
Date:   Thu Jun 26 16:00:44 2014 +0100

    string/memchr.c: Merge from gnulib

diff --git a/string/memchr.c b/string/memchr.c
index d1880cfbaed951a6..f4650672e01385de 100644
--- a/string/memchr.c
+++ b/string/memchr.c
@@ -57,10 +57,7 @@
 
 /* Search no more than N bytes of S for C.  */
 __ptr_t
-MEMCHR (s, c_in, n)
-     const __ptr_t s;
-     int c_in;
-     size_t n;
+MEMCHR (const __ptr_t s, int c_in, size_t n)
 {
   const unsigned char *char_ptr;
   const unsigned long int *longword_ptr;
